org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/hotspots/SecurityHotspotsHandlerServer.java

```diff
@@ -128,6 +143,94 @@ public void handle(ClassicHttpRequest request, ClassicHttpResponse response, Htt
     }
   }
 
+  private static class ShowHotspotRequestHandler implements HttpRequestHandler {
+
+    @Override
+    public void handle(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context)
+        throws HttpException, IOException {
+      if (!Method.GET.isSame(request.getMethod())) {
+        throw new MethodNotSupportedException("Only POST is supported");
+      }
+      ShowHotspotParameters parameters = getParameters(request, response);
+      if (parameters == null) {
+        return;
+      }
+
+      response.setCode(HttpStatus.SC_OK);
+
+      IConnectedEngineFacade connection = new BindingAssistant().bind(parameters.projectKey, parameters.serverUrl);
```

Comment:
This should maybe be executed in a background job, on not in the Handler, to not block the response for too long.